internal/pkg/chain/sampler.go

```diff
@@ -12,7 +12,20 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/crypto"
 )
 
-// A sampler draws randomness seeds from the chain.
+// Creates a new sampler for the chain identified by `head`.
+func NewSamplerAtHead(reader TipSetProvider, head block.TipSetKey) *SamplerAtHead {
+	return &SamplerAtHead{
+		sampler: NewSampler(reader),
+		head:    head,
+	}
+}
+
+// A sampler draws randomness seeds from the chain. The seed is computed from the minimum ticket of the tipset
+// at or before the requested epoch, mixed with the epoch itself (and is thus unique per epoch, even when they are
```

Comment:
I wouldn't mind something more precise: "randomness seeds are the black2b hash of the vrf digest of the minimum ticket of the tipset at or before the requested epoch concatenated with the big endian bytes of the epoch value"

Reply:
This is particularly useful because it makes it clear entropy bytes and dsts don't make it into the sampled bytes.

Reply:
I'll improve the comment, but sampling has nothing at all to do with entropy or DSTs. Those words do not appear in the chain package. I think mentioning them would be more confusing, and certainly prone to rot.

Reply:
Actually because MakeRandomSeed moved to `crypto`, I'll put the comment there.